robot_name: 'g1'
base_name: "pelvis"

state_topic: "g1/state"
command_topic: "g1/command"
change_controller_topic: "g1/change_controller"

reference_generator:
  type: "twist"
  twist:
    topic: "/cmd_vel"
  joystick:
    topic: "joy"
    ramped_velocity: 5.0
    x_index: 1
    y_index: 0
    yaw_index: 3
    x_scale: 0.5
    y_scale: 0.3
    yaw_scale: 0.2

g1_controller:
  hf_repo: "kubjonai/tbai-models"
  hf_model: "g1_rl_walking_model_unitree.onnx"
  policy_path: ""

  rate: 50.0

  joint_ids_map: [0, 6, 12, 1, 7, 13, 2, 8, 14, 3, 9, 15, 22, 4, 10, 16, 23, 5, 11, 17, 24, 18, 25, 19, 26, 20, 27, 21, 28]
  default_joint_pos: [-0.1, -0.1, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.3, 0.3, 0.3, -0.2, -0.2, 0.25, -0.25, 0.0, 0.0, 0.0, 0.0, 0.97, 0.97, 0.15, -0.15, 0.0, 0.0, 0.0, 0.0]
  action_scale: [0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25]
  stiffness: [100.0, 100.0, 100.0, 150.0, 40.0, 40.0, 100.0, 100.0, 100.0, 150.0, 40.0, 40.0, 200.0, 200.0, 200.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0, 40.0]
  damping: [2.0, 2.0, 2.0, 4.0, 2.0, 2.0, 2.0, 2.0, 2.0, 4.0, 2.0, 2.0, 5.0, 5.0, 5.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0]

  ang_vel_scale: 0.2
  joint_vel_scale: 0.05

static_controller:
  stand_controller:
    joint_angles: [
      -0.1,   # left_hip_pitch
      0.0,    # left_hip_roll
      0.0,    # left_hip_yaw
      0.3,    # left_knee
      -0.2,   # left_ankle_pitch
      0.0,    # left_ankle_roll
      # Right leg
      -0.1,   # right_hip_pitch
      0.0,    # right_hip_roll
      0.0,    # right_hip_yaw
      0.3,    # right_knee
      -0.2,   # right_ankle_pitch
      0.0,    # right_ankle_roll
      # Waist
      0.0,    # waist_yaw
      0.0,    # waist_roll
      0.0,    # waist_pitch
      # Left arm
      0.25,   # left_shoulder_pitch
      0.0,    # left_shoulder_roll
      0.0,    # left_shoulder_yaw
      0.97,   # left_elbow
      0.0,    # left_wrist_roll
      0.0,    # left_wrist_pitch
      0.0,    # left_wrist_yaw
      # Right arm
      0.25,   # right_shoulder_pitch
      0.0,    # right_shoulder_roll
      0.0,    # right_shoulder_yaw
      0.97,   # right_elbow
      0.0,    # right_wrist_roll
      0.0,    # right_wrist_pitch
      0.0     # right_wrist_yaw
    ]
  sit_controller:
    joint_angles: [
      0.0, 0.0, 0.0, 1.5, -0.8, 0.0,
      0.0, 0.0, 0.0, 1.5, -0.8, 0.0,
      0.0, 0.0, 0.0,
      0.5, 0.0, 0.0, 1.5, 0.0, 0.0, 0.0,
      0.5, 0.0, 0.0, 1.5, 0.0, 0.0, 0.0
    ]
  kp: 100
  kd: 5
  interpolation_time: 2.0
  rate: 50

g1_mimic_controller:
  rate: 50.0

  joint_ids_map: [0, 6, 12, 1, 7, 13, 2, 8, 14, 3, 9, 15, 22, 4, 10, 16, 23, 5, 11, 17, 24, 18, 25, 19, 26, 20, 27, 21, 28]
  default_joint_pos: [-0.302, -0.319, 0.00124, 0.000442, 0.00489, 0.00191, 0.00929, 0.00796, 0.00546, 0.672, 0.67, 0.2, 0.202, -0.368, -0.355, 0.194, -0.196, -0.00644, 0.00976, 0.00258, -0.00029, 0.605, 0.596, 0.00818, 0.00322, 0.00293, -0.00339, -0.00955, -0.00715]
  action_scale: [0.548, 0.548, 0.548, 0.351, 0.351, 0.439, 0.548, 0.548, 0.439, 0.351, 0.351, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.439, 0.0745, 0.0745, 0.0745, 0.0745]
  stiffness: [40.2, 99.1, 40.2, 99.1, 28.5, 28.5, 40.2, 99.1, 40.2, 99.1, 28.5, 28.5, 40.2, 28.5, 28.5, 14.3, 14.3, 14.3, 14.3, 14.3, 16.8, 16.8, 14.3, 14.3, 14.3, 14.3, 14.3, 16.8, 16.8]
  damping: [2.56, 6.31, 2.56, 6.31, 1.81, 1.81, 2.56, 6.31, 2.56, 6.31, 1.81, 1.81, 2.56, 1.81, 1.81, 0.907, 0.907, 0.907, 0.907, 0.907, 1.07, 1.07, 0.907, 0.907, 0.907, 0.907, 0.907, 1.07, 1.07]

g1_mimic_dance102:
  hf_repo: "kubjonai/tbai-models"
  hf_model: "g1_mimic_dance102_unitree.onnx"
  hf_motion_file: "G1_Take_102.bvh_60hz.csv"
  motion_fps: 60.0
  time_start: 2.8
  time_end: 24.0

g1_mimic_gangnam:
  hf_repo: "kubjonai/tbai-models"
  hf_model: "g1_mimic_gangnam_unitree.onnx"
  hf_motion_file: "G1_gangnam_style_V01.bvh_60hz.csv"
  motion_fps: 60.0
  time_start: 5.2
  time_end: 25.5

joint_names:
  - "left_hip_pitch_joint"
  - "left_hip_roll_joint"
  - "left_hip_yaw_joint"
  - "left_knee_joint"
  - "left_ankle_pitch_joint"
  - "left_ankle_roll_joint"
  - "right_hip_pitch_joint"
  - "right_hip_roll_joint"
  - "right_hip_yaw_joint"
  - "right_knee_joint"
  - "right_ankle_pitch_joint"
  - "right_ankle_roll_joint"
  - "waist_yaw_joint"
  - "waist_roll_joint"
  - "waist_pitch_joint"
  - "left_shoulder_pitch_joint"
  - "left_shoulder_roll_joint"
  - "left_shoulder_yaw_joint"
  - "left_elbow_joint"
  - "left_wrist_roll_joint"
  - "left_wrist_pitch_joint"
  - "left_wrist_yaw_joint"
  - "right_shoulder_pitch_joint"
  - "right_shoulder_roll_joint"
  - "right_shoulder_yaw_joint"
  - "right_elbow_joint"
  - "right_wrist_roll_joint"
  - "right_wrist_pitch_joint"
  - "right_wrist_yaw_joint"
